oligo_designer_toolsuite/oligo_specificity_filter/_filter_base.py

```diff
@@ -0,0 +1,58 @@
+import os
+from abc import ABC, abstractmethod
+
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+
+
+class SpecificityFilterBase(ABC):
+    """This is the base class for all specificity filter classes
+
+    :param dir_specificity: directory where alignement temporary files can be written
+    :type dir_specificity: str
+    """
+
+    def __init__(self, dir_specificity):
+        """Construnctor"""
+
+        self.dir_specificity = (
+            dir_specificity  # folder where we write the intermediate files
+        )
+        os.makedirs(self.dir_specificity, exist_ok=True)
+
+    @abstractmethod
+    def apply(self, oligo_DB, file_reference_DB, n_jobs):
+        """Apply filter to list of all possible probes in probe_info dictionary and filter out the probes which don't fulfill the requiremetns.
```

Comment:
...filter out the probes which don't **fulfil** the **requirements**.
        :param oligo_DB: database containing the probes **and** their features